components/resources/quizzes/careerQuiz/career-quiz.tsx

```diff
@@ -20,48 +33,72 @@ export enum Stage {
   RESULTS,
 }
 const CareerQuiz = () => {
-  const [saveUserPreferences] = useMutation(INSERT_CAREER_QUIZ_RESPONSE, {});
-  const exampleUserPreferences = [
-    {
-      degree: "Bachelor of Arts",
-      institution: "University of Waterloo",
-      name: "Angela",
-      email: "example@example.com",
-      industries: ["Advertising", "Design", "Fashion"],
-      skills: ["Writing code", "Writing", "Math"],
-      result: "Software Engineer",
-      tasks: ["Find trends in data"],
-    },
-  ];
+  const [degree, setDegree] = useState<string>("");
+  const [institution, setInstitution] = useState<string>("");
+  const [education, setEducation] = useState<EducationLevel | string>("");
+  const [experience, setExperience] = useState<string>("");
   // create results state object that
   // create custom type -- based on schema type in database
   const [stage, setStage] = useState<Stage>(Stage.START);
+  const [quizResponseId, setQuizResponseId] = useState<number>();
+
   const [triggerAnimation, setTriggerAnimation] = useState(true);
-  const [quizViewState, setQuizViewState] = useState<QuizViewState>();
-
-  useEffect(() => {
-    const quizViewState = {
-      title: quizData.title,
-      body: quizData.body,
-      questions: quizData.questions.map((question) => {
-        return {
-          title: question.title,
-          body: question.body,
-          options: question.options,
-        };
-      }),
-      currentQuestion: 0,
-      progress: 0,
-    };
+  const [quizViewState, setQuizViewState] = useState<QuizViewState>(
+    initializeQuizViewState
+  );
+  const [saveUserPreferences] = useMutation(UPSERT_CAREER_QUIZ_RESPONSE);
```

Comment:
check the comment I left on Jimi's PR but we only need the below mutation function and we can call it saveQuizResponse

you'll still have two handlers though